test/jdk/java/util/zip/ZipFile/InvalidBytesInEntryNameOrComment.java

```diff
@@ -21,163 +21,304 @@
-
+    @ParameterizedTest
+    @MethodSource("CENCommentOffsets")
+    public void testInValidEntryNameOrComment(int offset)
+            throws IOException {
+        createInvalidUTFEntryInZipFile(offset);
+        Throwable ex = assertThrows(ZipException.class, () ->
+                {
+                    try (ZipFile zf = new ZipFile(ZIP_FILE.toFile())) {};
+                }
+        );
+        assertEquals(CEN_BAD_ENTRY_NAME_OR_COMMENT, ex.getMessage());
     }
-
     /**
-     * Make a copy of the ZIP template and wrap it in a little-endian
-     * ByteBuffer
+     * Validate that a null is returned from ZipFile::getComment when the
+     * comment contains an invalid UTF-8 byte sequence.
+     * @throws IOException if an error occurs
      */
-    private ByteBuffer copyTemplate(byte[] template) {
-        return ByteBuffer.wrap(Arrays.copyOf(template, template.length))
-                .order(ByteOrder.LITTLE_ENDIAN);
+    @Test
+    public void testInValidZipFileComment() throws IOException {
+        createInvalidUTFEntryInZipFile(ZIP_FILE_COMMENT_OFFSET);
+        try (ZipFile zf = new ZipFile(ZIP_FILE.toFile())) {
+            assertNull(zf.getComment());
+        }
     }
 
     /**
-     * Make a ZIP with invalid bytes in the CEN comment field
+     * Validate that a ZipException is thrown when an entry name is encountered
+     * within a LOC contains an invalid UTF-8 byte sequence.
+     * @throws IOException if an error occurs
      */
-    private Path invalidComment(String name, byte[] template) throws IOException {
-        ByteBuffer buffer = copyTemplate(template);
-        int off = cenStart(buffer);
-        // Need to skip past the length of the name and extra fields
-        int nlen = buffer.getShort(off + NLEN);
-        int elen = buffer.getShort(off + ELEN);
-
-        // Comment field starts here
-        int coff = off + CEN_HDR + nlen + elen;
-
-        // Write invalid bytes
-        buffer.put(coff, INVALID_UTF8_BYTE_SEQUENCE, 0, INVALID_UTF8_BYTE_SEQUENCE.length);
-        return writeFile(name, buffer);
+    @Test
+    public void invalidZipInputStreamTest() throws IOException {
+        createInvalidUTFEntryInZipFile(LOC_FILE_HEADER_FILENAME_STARTING_OFFSET);
+        Throwable ex = assertThrows(ZipException.class, () ->
+                {
+                    try (ZipInputStream zis =
+                                 new ZipInputStream(new FileInputStream(ZIP_FILE.toFile()))) {
+                        zis.getNextEntry();
+                    };
+                });
+        assertEquals(LOC_BAD_ENTRY_NAME_OR_COMMENT, ex.getMessage());
     }
 
+    /**
+     * Utility method which modifies a Zip file starting at the specified
+     * offset to include an invalid UTF-8 byte sequence.
+     *
+     * @param offset starting offset within the Zip file to modify
+     * @throws IOException if an error occurs
+     */
+    private void createInvalidUTFEntryInZipFile(int offset) throws IOException {
```

Comment:
Is the "In" in this method name helpful? (To me it suggests a file argument, not a result) 
```suggestion
    private void createInvalidUTFEntryInZipFile(int offset) throws IOException {
```